### DISCUSSION TITLE
pdfcrop cannot be disabled?

### DISCUSSION THREAD

**Discussion by baptiste** [quarto-dev/quarto-cli#2917] [Q&A] [answered]
Referring to  #2861 (which I can't reopen), I think pdf figures in knitr are always cropped, 

https://github.com/quarto-dev/quarto-cli/blob/574396e68882c41336e9f539f4aee7ace6485f03/src/resources/rmd/execute.R#L272

The following MRE illustrates the problem:

````
---
title: "test"
format: pdf
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, crop=FALSE)
```

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
```{r}
library(ggplot2)
ggplot() + theme(plot.margin = margin(t=2,b=2,unit='cm')) # wide margins to see the effect (or lack thereof)
```
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

````

<img width="634" alt="Screenshot 2022-10-16 at 14 25 02" src="https://user-images.githubusercontent.com/18970/196013574-de0464cf-88f4-4d5d-b806-0e4a2a6a5908.png">

As a workaround I can set `knitr::knit_hooks$set(crop = function(...){})` to get the expected output

<img width="643" alt="Screenshot 2022-10-16 at 14 24 41" src="https://user-images.githubusercontent.com/18970/196013590-f277fa4d-2cd0-47fe-8b30-f3175be95815.png">

but one would expect `opts_chunk(crop=FALSE)` to prevent cropping without having to redefine the hook.

**Comment by mcanouil** [comment]
Do you mean like in the screenshot below (using Quarto v1.2.210)? (See <https://quarto.org/docs/computations/r.html#knitr-options> for details on how to set knitr options)
<img width="1624" alt="screenshot of Visual studio code with two panels. On the left a Quarto document where the execute echo is false and the knitr opts_chunk crop option is false in the yaml header, followed by some lorem ipsum and a ggplot2 code cell with big margins. On the right the rendered PDF showing the ggplot2 figures not croped" src="https://user-images.githubusercontent.com/8896044/196533251-443ede18-9af6-4bb1-8a5f-eca73e291d23.png">

<details><summary>See Quarto document content</summary>

````
---
title: "test"
format: pdf
execute:
  echo: false
knitr:
  opts_chunk:
    crop: false
---

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

```{r}
library(ggplot2)
ggplot() + theme(plot.margin = margin(t=2,b=2,unit='cm')) # wide margins to see the effect (or lack thereof)
```

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
````

</details>

EDIT: I can't reproduce what you are seeing, even when setting crop to true.

---

PS: can you change the category from "general" to "Q&A"? (from the desktop version of GitHub website)

**Reply by baptiste** [reply]
If you don't get cropping with `crop=TRUE`, I'm guessing you don't have pdfcrop installed on your machine?

**Comment by baptiste** [comment]
Hmm, strange, I'm getting this 

<img width="600" alt="Screenshot 2022-10-19 at 11 05 07" src="https://user-images.githubusercontent.com/18970/196553947-8a938bb8-3ec2-4b65-bde7-55844998ad25.png">

so I guess it's fixed in a different version of Quarto/Rstudio. I have Rstudio 2022.07.1 Build 554 and Quarto 1.1.251, on a Mac.

**Reply by mcanouil** [reply]
I'll have to double check, in theory I installed it, but since I changed a lot of things, I might have removed it recently.

**Reply by baptiste** [reply]
What do you get when you run `rmarkdown:::find_program("pdfcrop") != '' && tools::find_gs_cmd() != '' `? If FALSE, you cannot reproduce the issue.

**Comment by baptiste** [comment]
... actually, I just installed the pre-release Quarto 1.2.226, and I get the same cropped output (plus an glitch with lua filters for the author field)

**Reply by mcanouil** [reply]
What's the results with `rmarkdown` instead of Quarto?
Also, could you try using the proper Quarto syntax (as per my code example)?
Are you using the lastest version of `knitr`?

**Comment by salim-b** [comment]
I can fully reproduce the issue (using Ubuntu 22.04 and Quarto 1.3.361). If the `pdfcrop` binary and Ghostscript are available on the system, Quarto crops PDF figures regardless of the relevant chunk option.

I've tried to disable cropping without success by

- setting the chunk option (in YAML format) `crop: false`
- setting the chunk option (in YAML format) `fig-crop: false`
- setting `knitr.opts_chunk.crop: false` in `_quarto.yml`.
- setting `knitr.opts_chunk.fig_crop: false` in `_quarto.yml`. 
- setting `format.pdf.crop: false` in `_quarto.yml`.
- setting `format.pdf.fig_crop: false` in `_quarto.yml`.
- running `knitr::opts_chunk$set(crop = FALSE)` in a chunk before the figure chunks.
- running `knitr::opts_chunk$set(fig_crop = FALSE)` in a chunk before the figure chunks.

The only way that actually disabled PDF figure cropping was running `knitr::knit_hooks$set(crop = function(...){})` in a chunk before the figure chunks.

Note that I'm a bit confused about the correct name of the cropping chunk option on the knitr side of things: While the *R Markdown Cookbook* [mentions a `crop` chunk option](https://bookdown.org/yihui/rmarkdown-cookbook/crop-plot.html) (same what the [Quarto sources seem to refer to](https://github.com/quarto-dev/quarto-cli/blob/c8fb73c433215926d1b8b18f39e29bc1a215e33d/src/resources/rmd/execute.R#L277-L281)), the *R Markdown: The Definitive Guide* [mentions a `fig_crop` chunk option](https://bookdown.org/yihui/rmarkdown/pdf-document.html#figure-options-1). Quarto seems to ignore both of them.

**Reply by cscheid** [reply]
Can you please open a new issue? Thanks.

**Reply by salim-b** [reply, marked as answer]
See https://github.com/quarto-dev/quarto-cli/issues/5981
